pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -442,7 +442,7 @@ func (c *Cloud) AddSSHKeyToAllInstances(user string, keyData []byte) error {
 }
 
 // CurrentNodeName returns the name of the current node
-func (c *Cloud) CurrentNodeName(hostname string) (string, error) {
+func (c *Cloud) CurrentNodeName(hostname string) (types.NodeName, error) {
```

Comment:
@pmorie by that do you mean `var _ Interface = &Impl{}` or something else.  I'm inclined to agree with you, though not necessarily in this PR!

I think the original reasoning behind not just using checked casting was that in theory another object might implement one of the cloud provider interfaces.  I don't think anyone actually does that though.